### ISSUE TITLE
Assassin's Creed II (33230)

### ISSUE THREAD

**Issue by tr37ion** [ValveSoftware/Proton#190]
Starting the game shows a Steam dialog:

 `Failed to request product key from Steam, please try again in a few minutes.`

**Comment by veikk0** [comment]
Possibly related: https://github.com/ValveSoftware/steam-for-linux/issues/212

**Comment by jomarocas** [comment]
ok waiting for this issue for ac2

**Comment by ghost** [comment]
> ok waiting for this issue for ac2

The bug exists since 2012, Valve doesn't seem to care about this issue.
I bought AC2 to play with Proton and it doesn't work... not because Proton can't handle it, but because a bug in Steam itself... I'm super thrilled about this.

**Comment by jomarocas** [comment]
ok for me, the game working if you make this before starting steam 
`sed -i 's/legacykeyregistrylocation/legacykeyregistrylocation2/g' ~/.steam/steam/appcache/appinfo.vdf` you need to close steam if you have open and every time that you open steam

**Comment by Jpfonseca** [comment]
The bug still exists in version 4-11-4 of Proton and running the beta version of Steam Client of 15th of September.

**Comment by mikeant42** [comment]
bumping thread, has anyone come up with a solution? I've tried @jomarocas solution but it doesn't seem to fix it for me. Running elementary os. Odd thing is AC Odyssey, another uplay game works for me, but AC 2 doesn't.

**Comment by shoober420** [comment]
# Compatibility Report
- Name of the game with compatibility issues: Assassin's Creed II
- Steam AppID of the game: https://steamdb.info/app/33230/

## System Information
- GPU:  GTX 760
- Driver/LLVM version: NVIDIA 440.64
- Kernel version: 5.5.7-zen
- https://github.com/shoober420/linux-scripts/blob/main/home/shoober420/sysinfonvidia
- Proton version: 5.0-3

## Symptoms 

Without using the "sed -i 's/legacykeyregistrylocation/legacykeyregistrylocation2/g' ~/.steam/steam/appcache/appinfo.vdf" command in terminal before launching Steam, you will get this error message.

Failed to request product key from Steam, please try again in a few minutes.

VANILLA:
[steam-33230cdkey.log](https://github.com/ValveSoftware/Proton/files/4282407/steam-33230cdkey.log)

Using the "sed -i 's/legacykeyregistrylocation/legacykeyregistrylocation2/g' ~/.steam/steam/appcache/appinfo.vdf" command in terminal before launching Steam will spawn UPlay, which installs, you then create an account and/or log in. Then UPlay launches without recognizing Assassin's Creed II.

TERMINAL COMMAND FIRST:
[steam-33230first.log](https://github.com/ValveSoftware/Proton/files/4282377/steam-33230first.log)

After exiting UPlay and opening up Assassin's Creed II a second time, UPlay will ask you to link your Steam account. I personally don't feel comfortable doing this, so I chose not to do this. You are then greeted by a screen to enter your CD key.

This would be where you would enter the CD key that should display when launching the game, but its bypassed when using that terminal command at launch.

When starting up certain games on Steam, another window will open up showing your CD key. Assassin's Creed II seems to be the only game that has issues with this, and requires a command to be executed in terminal before launching Steam, which doesn't show the CD key screen at all, and just launches the game.

TERMINAL COMMAND SECOND:
[steam-33230second.log](https://github.com/ValveSoftware/Proton/files/4282379/steam-33230second.log)

**Comment by skutter-de** [comment]
@shoober420 Doesn't work for me. Same thing displays with or without the command

**Comment by shoober420** [comment]
@Gaming4LifeDE 

Do you have Steam set to run at startup? If so, make sure all Steam processes are killed, then execute the command in the terminal, and execute Steam from that same terminal.

**Comment by marco-gattei** [comment]
Distro: Linux Mint 19.3 Tricia
Kernel: 5.3.0-42-generic
GPU Driver: 3.0 Mesa 20.0.2 - kisak-mesa PPA
GPU: Intel UHD 620 (WHL GT2)
CPU:Intel Core i7-8565U @ 1.80GHz
PROTON: 5.0-5

Using the "sed -i 's/legacykeyregistrylocation/legacykeyregistrylocation2/g' ~/.steam/steam/appcache/appinfo.vdf" command in terminal before launching Steam  was working till 15/03/2020 ( i had 84 hours of play ).

After that the sed trick isn't working anymore. Steam was not running when i launched the script.

**Comment by shoober420** [comment]
Confirming "sed -i 's/legacykeyregistrylocation/legacykeyregistrylocation2/g' ~/.steam/steam/appcache/appinfo.vdf" no longer works for me as well.

**Comment by stanlog** [comment]
So sad, I tested this game on Proton today, I confirm, the workaround above doesn't work. Any other solution ?

**Comment by jomarocas** [comment]
> So sad, I tested this game on Proton today, I confirm, the workaround above doesn't work. Any other solution ?

only with the command sed before launch steam, will work, i tested in version before to proton 5.0.6 i suppose that this version no break nothing, i pass the game with version 4 you will try with this version

**Comment by stanlog** [comment]
I confirm that I'm pushing the command before launching Steam, same error, not immediately (Steam seems to looking for the key) then it fails.

My config:
Distro: Debian GNU/Linux bullseye/sid (64 bits)
Kernel: 5.5.0-2-amd64
GPU Driver: NVIDIA 440.82
GPU: NVIDIA GeForce GTX 960
CPU: Intel Core i5-6400 @ 2.70GHz 
PROTON: 5.0-6
Steam: steam-runtime_0.20200318.2

I pushed same report on ProtonDB.

**Comment by cverstege** [comment]
I've tried some more stuff, like changing the `vailidoslist` in `appinfo.vdf` to `oslist`, but it seems that steam always overwrites the file on launch or uses a freshly downloaded version if the immutable flag is set.

I've also tried adding the registry entries from a Windows install (including a valid key) to `registry.vdf` without success. See https://github.com/ValveSoftware/steam-for-linux/issues/212#issuecomment-725009666 for further details.

Sadly none of this had a positive effect.

**Comment by cverstege** [comment]
I think I found the cause of this Problem.

When installing Assassin's Creed 2 (https://steamdb.info/app/33230/) on Windows, it will only install the Windows CD Key DLC (https://steamdb.info/app/33361/), but not the Mac CD Key DLC (https://steamdb.info/app/33362/). It isn't even shown as an available DLC on the Windows Steam client.

In contrast, when installing the game on Linux, both CD Key DLCs are installed as well, but the Steam client can only fetch the Windows Key, but also wants to get the Mac key, as `ShowCDKeyOnLaunch` is set to true for both DLCs (see SteamDB links).

When launching the game on Windows, the Steam client will thus only try to get the Windows DLC Key, but the Linux client is trying to obtain the Windows and the Mac key on launch, which fails.

On Linux requesting the Windows Key with `steam steam://cdkeys/33361` works just fine. Requestig the Mac key with `steam steam://cdkeys/33362` will fail. Requesting the Mac Key on Windows will fail as well.
This is all expected behaviour I guess, as SteamPlay will only work with Windows titles.

So the problem here is that the steam Linux client wants to request all DLC keys when doing `steam steam://cdkeys/33230` but would only need the Windows DLC keys, as SteamPlay/Proton does only support running Windows Titles.
This is probably a problem with the steam Play configuration, as the Linux client will install the Mac only DLC with the Mac Key but then fails to get the key on startup.

I've also posted this information on https://github.com/ValveSoftware/steam-for-linux/issues/5649#issuecomment-727183485 and https://github.com/ValveSoftware/steam-for-linux/issues/5649#issuecomment-727183485.
Hopefully all the research I've done helps the Valve devs in fixing this problem.

**Comment by grktavlov** [comment]
Great info, @cverstege. It helped me find a workaround. By running [SteamEdit](https://steamedit.tg-software.com/) through Wine and changing the Mac DLC to Hidden,it is possible to start the game.

Make a link from `$WINEPREFIX/drive_c/Program Files (x86)/Steam/appcache/appinfo.vdf` to `$HOME/.local/share/Steam/appcache/appinfo.vdf` to edit the Steam for linux appinfo.vdf file.

I have not found any other way to change appinfo.vdf  without Steam overwriting it on startup.

**Comment by cverstege** [comment]
@grktavlov thanks for the tips. The tool actuallay just adds `steamedit^@^Bishidden` (taken from vim, I think `^@^B` represents a Byte with content 1?) to the DLC line inside `appinfo.vdf`.
To make it simple, we would just need to get this modificaton into a sed command. I'm not good with sed, especially when it comes to binary stuff like here present in the file.

I also needed to install the Crypt32 DLL via protontricks for Uplay to update (stuck at looking for patches loop).

Sidenote: When launching the game steam will crash when trying to fetch the DLC keys to show in the steam overlay. AC2 will still run fine though. AS a workaround just disable the ingame overlay for AC2.

**Comment by ljrk0** [comment]
More precisely it seems to edit after the last "entry"

```
...\@\H\H\H...
```
to
```
...\@\H\@\@\@\H\@steam_edit\@\Bis_hidden\@\A\@\@\@\H\H\H...
```

I assume that the sequence `\@\H\H\H` (0x00 0x8 0x8 0x8) means "end of element" and `\@\H\@\@\@\H\@` adds another "dictionary" to a list. The dictionary entry added in this case is:

```
\@\Bis_hidden\@\A\@\@
```
Where the `\@\B` is 0x00 0x2 and stands for "integer type", next comes the name, and then the four byte value which is 0x00 0x01 0x00 0x00, if the bytes are properly ordered this gives the value _1_.

Actually SteamEdit adds more entries as well, some of type `\@\A` (0x00 0x01) which is a string type, but they don't matter.

Using this information I came up with an even less invasive edit:  I simply changed this from
```
\@\Bshowcdkeyonlaunch\@\A\@\@
```
to
```
\@\Bshowcdkeyonlaunch\@\@\@\@
```
for the DLC.

**Comment by ljrk0** [comment]
SteamEdit does quite a lot of things. The VDF is basically a dictionary of nested dictionaries with the top-level key being the AppId.  The structure looks roughly as follows for the macOS DLC:

```json
{
  "appinfo": {
    "appid": 33362,
    "common": { },
    "extended": {
      "developer": "...",
      "dlcforappid": "...",
      "...": "...",
      "visibleonlywhensubscribed": 1
    },
    "steam_edit": {
       "is_hidden": 1,
       "base_name": "Assassin's Creed 2 - Mac",
       "base_type": "DLC"
    }
  }
}
```

Each scalar key has the format `<type> <name> \x00 <value> \x00`, so `is_hidden` looks like:

```
\x02is_hidden\x00\x00\x01\x00\x00
```

With \x02 (`\B`) designating the type `TYPE_INT32` and  `\x00\x01\x00\x00` being the value, i.e., 1.  A `TYPE_STRING` has \x01 (`\A). 

Dictionaries have `TYPE_DICT` with \x00 (`\@`), and furthermore, they don't use a \x00 byte to as termination symbol (being a list) but \x08 (`\H`). So they have the format `<type> <name> \x00 [entries] \x08` with `<type>` being \x00, being a `TYPE_DICT`.

So the above looks roughly like:

```
\x00
  \x00 appinfo \x00
    \x02 appid \x00 \x00\x51\x82\x00 \x00
    \x00 common \x00 ... \08
    \x00 extended\x00
      \x01 developer \x00 \x00
      \02 dlcforappid \x00 \x00\xCE\x81\x00 \x00
      ...
      \x02 visibleonlywhensubscribed \x00 \x00\x01\x00\x00 \x00
    \x08
    \x00 steam_edit \x00
       \x02 is_hidden \x00 \x00\x01\x00\x00 \x00
       \x01 base_name \x00 Assassin's Creed 2 - Mac \x00
       \x01 base_type \x00 DLC \x00
    \x08
  \x08
\x08
```

Actually, only the `is_hidden` entry is really needed.

**Comment by pdfrod** [comment]
Thanks a lot for your tips guys; I was finally able to get Assassin's Creed II running!

Updating the appinfo.vd file is not that simple because the entries have a checksum. If the checksum is not correct, Steam will simply wipe out the file and start a new one. I wasn't able to figure out how the checksum is calculated, but SteamEdit is able to generate them correctly. So I grabbed SteamEdit's output to generate a very minimal [appinfo.vdf](https://gist.github.com/pdfrod/e326c316426ced1d2abec86425443362) with a single entry ("Assassin's Creed 2 - Mac").

Simply replacing the existing appinfo.vdf with this one does the job - Steam will append new game entries but leave the existing "Assassin's Creed 2 - Mac" entry untouched.

**Comment by tralph3** [comment]
Thank god for this thread. It was very helpful in my research. I'm currently making a clone of Steam Edit, it will have more features and be completely cross-platform, also open source. I will also leave an explanation on how the appinfo.vdf file works in the README.

Tim Green (the creator of Steam Edit) gave me a hand in a couple of situations and explained how the checksum works. Basically, you need to format the modified data in a specific way and pass it through sha1, more details once I release this... hopefully soon.

**Comment by ljrk0** [comment]
> Thank god for this thread. It was very helpful in my research. I'm currently making a clone of Steam Edit, it will have more features and be completely cross-platform, also open source. I will also leave an explanation on how the appinfo.vdf file works in the README.
> 
> Tim Green (the creator of Steam Edit) gave me a hand in a couple of situations and explained how the checksum works. Basically, you need to format the modified data in a specific way and pass it through sha1, more details once I release this... hopefully soon.

That's great to hear as it gives me a reason less to procrastinate my thesis with doing what you do :D 

Would you be imterested to share the current state of the source already (no matter how dirty)? I think there are quite a few people in here who are interested to help!

**Comment by tralph3** [comment]
> 
> That's great to hear as it gives me a reason less to procrastinate my thesis with doing what you do :D
> 
> Would you be imterested to share the current state of the source already (no matter how dirty)? I think there are quite a few people in here who are interested to help!

This is quite a bad time for that since I'm doing a major rewrite. I found out the module I was using produces a vdf file that Steam rejects. I tried to modify the binary data manually and it worked. The application wasn't designed to work with bytes directly, so I'm currently figuring out how everything works and how to handle it to adapt the application (hence, why this post was useful).

**Comment by tralph3** [comment]
I finally released my program. Here you go: https://github.com/tralph3/Steam-Metadata-Editor

**Comment by Killroystyx** [comment]
Hey all, im not sure exactly where to go for help on this but I've got an issue with the game that is really hard for me to troubleshoot. Everything i try searching gives me really dumb game walkthroughs on how to equip weapons or an unanswered bug report from a decade ago which likely doesnt apply to me.

I have AC2 running in a wacky way. Native steam > lutris > wine steam > ac2. Its been going pretty well so far. I couldn't figure out how to get my steam controller to relay xinput to the game but keyboard and mouse inputs can be changed just fine via native steam BPM.

The issue is the weapon wheel. I can see it and hold it open with whatever key i assign to it. I can use mouse input to highlight the desired item. But releasing the weapon wheel button to select that item never works. 

I've reached the mission called Knowledge is power, where you receive the pistol mod and are tasked to shooy some dummies. This is as far as i can go until i figure out what is going on.

Fyi the reason im playing like this is bc i have not had success following the vdf file replacement method. Although google was also hiding this page from me and i had to get here via the link from the proton db. So i have yet to read all the details here and was following directions from other sources.

**Comment by mberndt123** [comment]
I'm having this problem too, please fix. We've paid good money for our games, we deserve to play them.

**Comment by Managor** [comment]
Now I feel silly for pestering Ubisoft support with this.

**Comment by smintman** [comment]
This issue has been going for 2 years now, seems like it should be quite easy for Valve to fix the CD key issue? once I got over the CD key issue had issues with mising uplay client and uplay can not find the downloaded game files, put my list steps to fix here: 

https://www.protondb.com/app/33230

Surprised I need to do these fixes on such an old and popular game, feels like these things should be relatively easy to fix, bit worried that they have not been fixed and does not bode well for Steam decks "most games just work" statement!?! or am I missing something?

**Comment by cverstege** [comment]
I hope this issue gets some attention from Valve once the Steam Deck is released, as AC II and AC Brotherhood are part of the all time best games people will want to play on the Deck. Especially the key bug should be an easy fix, I guess.
The Uplay issues you encountered are not the same for me, though the uplay update process during the first launch does not work perfectly. I need to stop the game through steam after getting an error from uplay and then relaunch it. During the second launch uplay finally updates and the game launches. But uplay does not Download the game again for me.

**Comment by pdfrod** [comment]
> This issue has been going for 2 years now, seems like it should be quite easy for Valve to fix the CD key issue? once I got over the CD key issue had issues with mising uplay client and uplay can not find the downloaded game files, put my list steps to fix here:
> 
> https://www.protondb.com/app/33230
> 
> Surprised I need to do these fixes on such an old and popular game, feels like these things should be relatively easy to fix, bit worried that they have not been fixed and does not bode well for Steam decks "most games just work" statement!?! or am I missing something?

I feel you man. An issue that could be probably be solved in 10 minutes by Valve, and we are condemned to live with it forever simply because it's far down on their priority list.

If they would point me at the code, I'd gladly fix the problem myself.

**Comment by thaewrapt** [comment]
That looks more of a configuration issue, in which case it may be still Ubisoft's move rather than Valve's.

**Comment by tralph3** [comment]
Yup, this is Ubi's job to fix it. Which means it won't get fixed, and you better get used to patching it yourself.

**Comment by cverstege** [comment]
I don't agree with you here. Just take a look at SteamDB, everything is configured just fine. Even the Mac DLC is configured Mac only (https://steamdb.info/app/33362/).
This game works out of the Box on Windows. You can't access the Mac DLC (key) on Windows at all, because it's for MacOS. So why on earth, would steam on Linux request the Mac Key and fail doing so, if I play this game via a Windows compatibility layer. And why on earth is steam installing the Mac DLC on Linux if it's Mac only? It does not install it on Windows (yes I've tested it). Steam should just show the keys available for Windows when running on Linux.

As long as there is no MacOS compatibility layer, there is no reason at all, that keys or DLCs only available on MacOS should be shown on Linux.
Even if Valve wanted to show all the Keys on Linux, they could just make it work instead of throwing an error and preventing the game from launching.
When I run steam on MacOS, I'm fairly certain that I can only access the Mac Key and no Windows keys at all (did not test this, as I don't have MacOS). And it probably just works fine on MacOS as well.

So why is this an issue on Ubisoft's side and not on Valve's? This is an issue with the Steam Client for Linux imo.

**Comment by smintman** [comment]
I've tried to semi automate the steps I have to follow to get the assassins creed 2 and brotherhood installed and working. You can find them here, maybe it will help someone https://github.com/smintman/assfix 

Automates downloading the CD Key fix and installs updated version of ubisoft connect.

**Comment by ranplayer** [comment]
Summarizing everything that has been posted on this topic, here is a little step-by-step about running Assassin's Creed II with **Proton 6.3**:

1) After installing and running the game for the first time, a popup will show up with a message related to CDKey. Then the game exists automatically.
1.1) Why does this happen? Because the Steam client on Linux tries to load the DLC available for MacOS.
2) Optional step: replacing the **UPlay** client by the **UbisoftConnect** client (to avoid updates and headaches)
2.1) Download the UbisoftConnect installer (`UbisoftConnectInstaller.exe`) from the official website https://ubisoftconnect.com
2.2) Remove the `UplayInstaller.exe` file from the game's installation folder (usually located at `~/.steam/steam/steamapps/common/Assassin's Creed 2`)
2.3) Move the `UbisoftConnectInstaller.exe` file to the game's installation folder and rename it to `UplayInstaller.exe`
3) Close the Steam client
4) Replace the **appinfo.vdf** file located in `~/.steam/steam/appcache` by the following https://gist.github.com/pdfrod/e326c316426ced1d2abec86425443362/raw/cb0c38bd15a58d1ccc1794078b384d3c37b691b7/appinfo.vdf
4.1) Explanation: the downloaded file contains only the metadata that disables the MacOS DLC (Steam will update this file on next boot, but will keep the metadata related to disabling the DLC)
5) Restart the Steam client
6) Start the game and have fun!

**Extras tips**
- For better performance: disable the UbisoftConnect overlay.
- Gamepad not working: Disable the Steam overlay and the Steam input settings for this game.

I hope this reaches other people wanting to play the game on Linux since neither Ubisoft nor Valve seem to be prioritizing this issue. A huge thank you for all the people that have contributed to make this classic playable.

**Comment by smintman** [comment]
Nice summary, uplay played nicely for me on latest experimental. If only they could fix the client getting confused and trying to request the Mac CDKey!! It would just work! Valve are you listening!?!!

**Comment by smintman** [comment]
After doing a clean install I've discovered another thing that is causing launch issues, I was getting the can not find uplay issue, and Iooked like the first run stuff just wasn't happening. The reason behind this is that the game needs the Steam common re-distributables installed which the current install doesn't appear to download (I think the dev has to opt in)

If you have installed a game that a dev has opted in the re-distributables are already installed and all works ok..

I've created a script that automates the following, there is a menu now that you just need to select if you want to fix assassins creed 2 or brotherhood

https://github.com/smintman/assfix

1) Downloads either the assasasins creed 2/brotherhood hacked vdf from github and copies in to place
2) Opens up steam and downloads to common re-distributables depot
3) If brotherhood is selected, it will download the updated uplay installed and copy into place so that when the game is run it will it use that instead. I've only gone this for brotherhood because on first run it tries to update uplay but throws an error (it actually has updated and on second run all is fine) but I thought that I will add in the updated installer to avoid the error. Assassins creed 2's version of uplay updates itself just fine.

None of these work arounds would be needed if Ubisoft/Valve did the required work, hopefully someone is on the case.. somewhere... maybe?

**Comment by ghost** [comment]
Apologies for bumping the thread. 

Tried the method put forward by @ranplayer, which didn't work. Another forum discussing this topic pointed out that the error is sometimes produced due to Steam not installing the "**Steam Common Redistributables**". Under the presumption that my Steam install might have some problems, I performed a fresh re-install of Steam. After that, I installed Portal 2 - a Valve game - hoping it would install the redistributables with it, which it did.

Then I repeated the method, and it worked.

**Comment by smintman** [comment]
I think this affects lots of games. Especially when on fresh install of steam.  
Was so confused when first started trying to get games working as to why instructions that others had used were not working. 
I think valve need to be a bit more proactive with devs helping them get the right set up to work on Linux. As I guess some just don’t know.

**Comment by begin-theadventure** [comment]
~~https://github.com/begin-theadventure/acfix
I created another way to apply the fix, which I think is better because you can add it to multiple ACs at once - practically unlike with .vdf - theoretically well unless you wanna share your entire library...
+I added fixes for Unity and Brotherhood Standard Edition.~~ Not needed anymore.

**Comment by smintman** [comment]
The issue might be fixed, I updated my client (I'm on beta channel) and deleted the prefix and deleted the appcache.vdf (to clear any previous fixes) and I managed to start, update and play assassins creed 2 deluxe and brotherhood!
Also can request CD key without error!

Can others try to check the issue is resolved for them!!

**Comment by lukassup** [comment]
@smintman confirmed :+1:

**Comment by lukassup** [comment]
Maybe worth mentioning that I could not start a new game when Ubisoft Connect launcher was in offline mode

**Comment by begin-theadventure** [comment]
I forgot to reply but it does indeed work now.

**Comment by NexonSU** [comment]
So, right now we have only this problems:
- Game crash on Alt+Tab. Fixed by using virtual desktop mode in wine, but with this, mouse cursor always captured by game. 
- Ridiculously low LOD distance and low shadow resolution. On windows it can be fixed via cheatengine hacks.

Maybe anyone find a way?

**Comment by tralph3** [comment]
For alt+tab, try using gamescope. For the LOD, see if you can change that in the game's config files.

**Comment by NexonSU** [comment]
> For alt+tab, try using gamescope. For the LOD, see if you can change that in the game's config files.

Gamescope sounds like ultimate solution to solve this kind of problems. It's still not working with NVIDIA, but it will soon.
Anyway, right now I using wine virtual desktop + MouseWarpOverride=disable + DS4, it works.

About LODs, game doesn't allow to change that via config file.

**Comment by tralph3** [comment]
Well, for Linux you can use https://github.com/scanmem/scanmem as a replacement for Cheat Engine. It's the same thing, a memory editor. You will need to find the appropriate addresses and make the needed changes.

**Comment by NexonSU** [comment]
Yes, I tried scanmem via GameConqueror and I can't find any addresses with constant 256, 512, 1024 or 2048 values.
Also, addresses from "AC2 Visual Fixup Mod by #Bucket5" not working, and as AC2VFM.EXE itself.

**Comment by begin-theadventure** [comment]
> > For alt+tab, try using gamescope. For the LOD, see if you can change that in the game's config files.
> 
> Gamescope sounds like ultimate solution to solve this kind of problems. It's still not working with NVIDIA, but it will soon. Anyway, right now I using wine virtual desktop + MouseWarpOverride=disable + DS4, it works.
> 
> About LODs, game doesn't allow to change that via config file.

I also noticed a problem with alt+tab and read on [ProtonDB](https://www.protondb.com/app/33230) [RADA77435] about virtual desktop mode, I enabled it by:
1] opening `pathto/SteamLibrary/steamapps/compatdata/33230/pfx` in the terminal
2] running `export WINEPREFIX=$(pwd); winecfg`
However the mouse didn't work outside of it and here I read your comment mentioning MouseWarpOverride.
I did as before but instead ran:
`winetricks mwo=disable` 
and now everything's fine.
Thanks!

**Comment by begin-theadventure** [comment]
> Gamescope sounds like ultimate solution to solve this kind of problems. It's still not working with NVIDIA, but it will soon. Anyway, right now I using wine virtual desktop + MouseWarpOverride=disable + DS4, it works.
>
> About LODs, game doesn't allow to change that via config file.

I also noticed a problem with alt+tab and read on [ProtonDB](https://www.protondb.com/app/33230) [RADA77435] about virtual desktop mode, I enabled it by:
1] opening `pathto/SteamLibrary/steamapps/compatdata/33230/pfx` in the terminal
2] running `export WINEPREFIX=$(pwd); winecfg`
3] checking `Emulate a virtual desktop` under `Graphics` typing native resolution under `Desktop size` clicking `Apply` and `OK`
However the mouse didn't work outside of it and here I read your comment mentioning MouseWarpOverride.
I did as before but instead ran:
`winetricks mwo=disable` 
and now everything's fine.
Thanks!
//Edit: After tabbing out of the game/virtual desktop it still captures mouse movements and clicks so my solution is to stop the game and continue via e.g. system monitor.

**Comment by NexonSU** [comment]
@begin-theadventure glad to hear. But, next time try to use protontricks, it allows you to select game in GUI:
<details>
  <summary>Screenshot</summary>
  <img src="https://user-images.githubusercontent.com/5037786/175777044-bf6964e1-1b3c-4efa-aafe-51aaa0e6b8b2.png">
</details>
BTW, I completed my AC2 walkthrough like this. It's still awesome game.

**Comment by Sterophonick** [comment]
Unfortunately, it appears this game's controller support has been broken as of Proton 8.0. It worked correctly in Proton 7.x.

Playing on the Steam Deck, external controllers also do not work, and keyboard emulation is not an option because of the camera.

[steam-33230.log](https://github.com/ValveSoftware/Proton/files/11256239/steam-33230.log)

**Comment by kisak-valve** [comment]
Hello @Sterophonick, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by Sterophonick** [comment]
> Hello @Sterophonick, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

Done.
https://gist.github.com/Sterophonick/edf16939772f95baa16caac3561c3264

**Comment by Sterophonick** [comment]
> Unfortunately, it appears this game's controller support has been broken as of Proton 8.0. It worked correctly in Proton 7.x.
> 
> Playing on the Steam Deck, external controllers also do not work, and keyboard emulation is not an option because of the camera.
> 
> [steam-33230.log](https://github.com/ValveSoftware/Proton/files/11256239/steam-33230.log)

Caused by weird prefix creation after using `Delete Proton Files` on the Steam Deck. Fixed with https://github.com/ValveSoftware/Proton/commit/576a839c18e636f017ef301224c536a9e689bad8.

**Comment by alasky17** [comment]
@Sterophonick I'm looking into this, and unable to reproduce the failure you are seeing.  First - did you use the "Delete Proton files" option?  Several folks used this when upgrading to 8.0-1, and unfortunately there is a SteamOS bug that was causing controllers to break.  We have worked around this temporarily with the 8.0-1d release which is now live.  If this is the case, you should be able to repair the problem by using "Delete Proton files" again and then launching on 8.0-1d again.  If this does not work and you did use "Delete Proton files", I think it is worthwhile for you to try manually deleting the prefix before doing any other investigation.

---------------------------------
These questions are for follow up in the event that I was wrong about my hunch regarding Delete Proton files.  Please disregard if the instructions above fix your error (but it would be great if you let me know) :) 

I'm assuming that controllers do not work at all, even on the menus for you starting with 8.0-1? 

What kind of controller are you using? Also, could you go into the controller settings and check the name of the controller layout(s) for each controller (including the built-in steam controller)?  

Also, if you haven't already done this, could you switch the proton version back to 7.0-6 to make sure the controllers are still working for you there?

**Comment by Sterophonick** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/190#issuecomment-1514986837

Swapping back to 7.0 actually fixed it. Proton worked around the SteamOS bug with that commit, so `Delete Proton Files` now works correctly. The problem affected other games, in more ways than just XInput.

Everything is all fine now.

**Comment by kisak-valve** [comment]
Hello @Sterophonick, unfortunately, that response is ambiguous. I could mean that you switched to Proton 7.0-6 and called it good enough, or regenerated the game's wineprefix with Proton 7.0-6 and switched back to Proton 8.0-1 and called it good enough. 

Please make it explicitly clear if you tested having Proton 8.0-1d regenerate the game's wineprefix and then checked the game's behavior so that we can drop the regression label.

**Comment by Sterophonick** [comment]
> Hello @Sterophonick, unfortunately, that response is ambiguous. I could mean that you switched to Proton 7.0-6 and called it good enough, or regenerated the game's wineprefix with Proton 7.0-6 and switched back to Proton 8.0-1 and called it good enough. 
> 
> Please make it explicitly clear if you tested having Proton 8.0-1d regenerate the game's wineprefix and then checked the game's behavior so that we can drop the regression label.

Yes, I have tested having Proton 8.0-1d regenerate the prefix.

**Comment by L7vanmatre** [comment]
After the most recent Ubisoft update, it doesn't work via any Proton version. 

However, not even Proton Experimental works either. It just boots, stays trying to boot there for a bit and says it's in-game, and then closes.

<details><summary>Here are the log files of opening Steam via console.</summary>

```
fsync: up and running.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
/bin/sh\0-c\0/home/gary/.steam/debian-installation/ubuntu12_32/reaper SteamLaunch AppId=33230 -- /home/gary/.steam/debian-installation/ubuntu12_32/steam-launch-wrapper -- '/home/gary/.steam/debian-installation/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point --verb=waitforexitandrun -- '/home/gary/.steam/debian-installation/steamapps/common/Proton - Experimental'/proton waitforexitandrun  '/home/gary/.steam/debian-installation/steamapps/common/Assassin'\''s Creed 2/AssassinsCreedIIGame.exe' -uplay_steam_mode\0
chdir "/home/gary/.steam/debian-installation/steamapps/common/Assassin's Creed 2"
ERROR: ld.so: object '/home/gary/.steam/debian-installation/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/gary/.steam/debian-installation/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
ERROR: ld.so: object '/home/gary/.steam/debian-installation/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/gary/.steam/debian-installation/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/gary/.steam/debian-installation/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
WARNING: discarding _NET_WM_PID 5 as invalid for X11 window - use specialized XCB_X11_TO_PID function!
WARNING: discarding _NET_WM_PID 5 as invalid for X11 window - use specialized XCB_X11_TO_PID function!
WARNING: discarding _NET_WM_PID 5 as invalid for X11 window - use specialized XCB_X11_TO_PID function!
WARNING: discarding _NET_WM_PID 5 as invalid for X11 window - use specialized XCB_X11_TO_PID function!
fsync: up and running.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
Setting breakpad minidump AppID = 33230
Steam_SetMinidumpSteamID:  Caching Steam ID:  76561198084095266 [API loaded no] (NOTE: This is the line it takes the longest to be at.)
[2023-08-01 22:27:02] Background update loop checking for update. . .
[2023-08-01 22:27:02] Checking for available updates...
[2023-08-01 22:27:02] Downloading manifest: https://client-update.akamai.steamstatic.com/steam_client_publicbeta_ubuntu12?t=2327939811
[2023-08-01 22:27:02] Manifest download: send request
[2023-08-01 22:27:02] Manifest download: waiting for download to finish
[2023-08-01 22:27:03] Manifest download: finished
[2023-08-01 22:27:03] Download skipped: /steam_client_publicbeta_ubuntu12?t=2327939811 version 1690583737, installed version 1690583737, existing pending version 0
[2023-08-01 22:27:03] Nothing to do
pid 6306 != 6305, skipping destruction (fork without exec?)
Game 33230 created interface STEAMAPPLIST_INTERFACE_VERSION001 / 
Game 33230 created interface STEAMAPPS_INTERFACE_VERSION008 / 
Game 33230 created interface STEAMHTMLSURFACE_INTERFACE_VERSION_005 / 
Game 33230 created interface STEAMHTTP_INTERFACE_VERSION003 / 
Game 33230 created interface STEAMINVENTORY_INTERFACE_V003 / 
Game 33230 created interface STEAMMUSICREMOTE_INTERFACE_VERSION001 / 
Game 33230 created interface STEAMMUSIC_INTERFACE_VERSION001 / 
Game 33230 created interface STEAMPARENTALSETTINGS_INTERFACE_VERSION001 / 
Game 33230 created interface STEAMREMOTESTORAGE_INTERFACE_VERSION014 / 
Game 33230 created interface STEAMSCREENSHOTS_INTERFACE_VERSION003 / 
Game 33230 created interface STEAMUGC_INTERFACE_VERSION015 / 
Game 33230 created interface STEAMUSERSTATS_INTERFACE_VERSION012 / 
Game 33230 created interface STEAMVIDEO_INTERFACE_V002 / 
Game 33230 created interface SteamController008 / 
Game 33230 created interface SteamFriends017 / 
Game 33230 created interface SteamInput002 / 
Game 33230 created interface SteamMatchGameSearch001 / 
Game 33230 created interface SteamMatchMaking009 / 
Game 33230 created interface SteamMatchMakingServers002 / 
Game 33230 created interface SteamNetworking006 / 
Game 33230 created interface SteamUser021 / 
Game 33230 created interface SteamUser021 / User
Game 33230 created interface SteamUtils010 / 
Game 33230 method call count for IClientAppManager::GetAppStateInfo : 1
Game 33230 method call count for IClientUtils::RecordSteamInterfaceCreation : 27
Game 33230 method call count for IClientUtils::GetAPICallResult : 1
Game 33230 method call count for IClientUtils::GetAppID : 29
Game 33230 method call count for IClientUser::GetEncryptedAppTicket : 1
Game 33230 method call count for IClientUser::RequestEncryptedAppTicket : 1
Game 33230 method call count for IClientUser::RequestLegacyCDKeyFromApp : 1
Game 33230 method call count for IClientUser::GetSteamID : 2
Uploaded AppInterfaceStats to Steam
```

</details>

I don't know why the newlines don't show it correctly in actual visual output. I'm not well-versed in Github unfortunately. However, this is what is there. I hope it's helpful for something. Here's a Pastebin link for easier reading: https://pastebin.com/VHf1rAPH

**Comment by alasky17** [comment]
@L7vanmatre The breakage from the most recent Ubisoft update can be fixed by just adding "WINE_SIMULATE_WRITECOPY=1 %command%" to your launch options.  Could you test an older Proton version that used to work before the most recent update (like 8.0-3) with this command to see if the game was broken by an experimental update?  If you get the same failure, could you please get a log with PROTON_LOG=1 %command% on experimental?  

Also, please also copy your system information from Steam (Steam -> Help -> System Information and Steam -> Help -> Steam Runtime Diagnostics) and put each in a [gist](https://gist.github.com/), then include a link to the gists in this issue report.

**Comment by kisak-valve** [comment]
Trying to play Assassins Creed 2 on Steam

Issue transferred from https://github.com/ValveSoftware/Proton/issues/7507.
@DrThomasXLight posted on 2024-02-19T03:34:48:

Hi,

I'm trying to play Assasins Creed II on my PC using Linux, but every time I launch the game, I receive this uplay window and I get stuck in it even when my credentials to ubisoft are correct.

How can I play this Game? I enjoy this game in PS4 and would like to play it again on STEAM now.

My System info:

# System Details Report
---

## Report details
- **Date generated:** 2024-02-17 19:28:08

## Hardware Information:
- **Hardware Model:** ASUS TUF GAMING B560M-PLUS WIFI
- **Memory:** 16.0 GiB
- **Processor:** Intel® Core™ i5-10400F × 12
- **Graphics:** NVIDIA GeForce GTX 1650
- **Disk Capacity:** 1.2 TB

## Software Information:
- **Firmware Version:** 0811
- **OS Name:** openSUSE Tumbleweed
- **OS Build:** (null)
- **OS Type:** 64-bit
- **GNOME Version:** 45.3
- **Windowing System:** X11
- **Kernel Version:** Linux 6.7.4-1-default
- 
![Screenshot from 2024-02-17 19-20-24](https://github.com/ValveSoftware/Proton/assets/56457861/97c5aa10-8c3b-4b9f-98b9-6289e8b99be3)
![Screenshot from 2024-02-17 19-20-24](https://github.com/ValveSoftware/Proton/assets/56457861/5e25ffa0-f621-4e57-8de0-91a8a393f027)

**Comment by gverm** [comment]
@DrThomasXLight I was experiencing the same issue and solved it by installing Ubisoft Connect inside the Assassins Creed 2 prefix. I used protontricks for this.

**Comment by DrThomasXLight** [comment]
Let me try that.. I'll post how it goes.. wish me luck 👍

**Comment by Solandis** [comment]
@gverm could you explain how to do it? 

> I was experiencing the same issue and solved it by installing Ubisoft Connect inside the Assassins Creed 2 prefix. I used protontricks for this.

**Comment by DrThomasXLight** [comment]
Dude, It worked for me, but depends of the protontricks version you use.
It need to allow you to install the latest Ubisoft connect exe, did it
yesterday inside the AC2 proton config (bottle). After it updates, you cab
run AC2 as it should,  you connect with your Ubisoft credentials, it
updates game stuff and voila.. AC2 starts 🙌

That was my experience and I'm enjoying again one of my favorite games in
history..  thanks to @gverm for the tip.. nice tip indeed
